### ISSUE TITLE
ImageDataGenerator return dtype vs  ImageDataGenerator.flow_from_dataframe dtype mismatch

### ISSUE THREAD

**Issue by SaiSandeepKantareddy** [keras-team/keras#16763]
Mismatch of dtypes and altering the image after reading the function(flow_from_dataframe), 

ImageDataGenerator(preprocessing_function = preprocess,)

this gives:

<img width="259" alt="Screen Shot 2022-07-07 at 12 10 52 AM" src="https://user-images.githubusercontent.com/16205133/177696596-cdc51507-6018-46e0-a4b2-dcea613a62ae.png">

train_generator = train_data_generator.flow_from_dataframe(
    dataframe = df_train,
    directory = ‘direct'
    x_col= 'path',
    y_col= ‘’column'
    batch_size = 32,
    seed = 42,
    shuffle = True,
    class_mode= 'raw',
    flip_vertical = True,
    #color_mode = 'rgb'
    target_size = (img_size, img_size))

this gives:

<img width="263" alt="Screen Shot 2022-07-07 at 12 11 17 AM" src="https://user-images.githubusercontent.com/16205133/177696526-82bf0a51-6e89-41bc-917a-16722279d519.png">

**Comment by sushreebarsa** [comment]
@SaiSandeepKantareddy 
In order to reproduce the issue reported here, could you please provide the complete code, the dataset and tensorflow version you are using. Thank you!

**Comment by SaiSandeepKantareddy** [comment]
Tensorflow Version: 2.3.0 Keras 2.4.0, actually the data I am using is confidential, but the error is like when you pass the data like the first image after using the preprocess function: I am getting this just black and white pixels alone but when I am using the same preprocess function on a single image I cannot see any problem,

preprocess function code:

def preprocess(image):
#         img=cv2.imread(df_train['path'].iloc[1])
        img1=image.astype('uint8')
        img= np.array(PIL.Image.fromarray(img1).convert('L'))
            # Otsu's thresholding after Gaussian filtering
        if np.mean(img[:,:255])<np.mean(img[:,img.shape[1]-255:]):
            img=cv2.flip(img,1)
        blur = cv2.GaussianBlur(img,(5,5),0)
        #print(blur)
        _, breast_mask = cv2.threshold(blur,0,255,cv2.THRESH_BINARY+cv2.THRESH_OTSU)

        cnts, _ = cv2.findContours(breast_mask.astype(np.uint8), cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
        cnt = max(cnts, key = cv2.contourArea)
        x, y, w, h = cv2.boundingRect(cnt)

        image=np.array(PIL.Image.fromarray(img[y:y+h, x:x+w]).convert('RGB'))
        image=cv2.resize(image,(256,256))
        #image1=image.astype('uint8')
        return image
        #return tf.keras.applications.xception.preprocess_input(image)

**Comment by sushreebarsa** [comment]
@SaiSandeepKantareddy Thank you for the update!
Could you please upgrade to the latest TF version 2.9.0 and let us know if the issue still persists ?
Thank you!
